lib/ethon/easy/queryable.rb

```diff
@@ -95,18 +96,37 @@ def mime_type(filename)
       def recursively_generate_pairs(h, prefix, pairs)
         case h
         when Hash
-          h.each_pair do |k,v|
-            key = prefix.nil? ? k : "#{prefix}[#{k}]"
-            pairs_for(v, key, pairs)
-          end
+          encode_hash_pairs(h, prefix, pairs)
         when Array
-          h.each_with_index do |v, i|
-            key = "#{prefix}[#{i}]"
-            pairs_for(v, key, pairs)
+          if rack_arrays
+            encode_rack_array_pairs(h, prefix, pairs)
+          else
+            encode_indexed_array_pairs(h, prefix, pairs)
           end
         end
       end
 
+      def encode_hash_pairs(h, prefix, pairs)
+        h.each_pair do |k,v|
+          key = prefix.nil? ? k : "#{prefix}[#{k}]"
+          pairs_for(v, key, pairs)
+        end
+      end
+
+      def encode_indexed_array_pairs(h, prefix, pairs)
+        h.each_with_index do |v, i|
+          key = "#{prefix}[#{i}]"
+          pairs_for(v, key, pairs)
+        end
+      end
+
+      def encode_rack_array_pairs(h, prefix, pairs)
```

Comment:
That would imply adding `Rack` as a dependency. I personally always try to depend only on what is really strictly necessary. Considering also that it is unlikely `Rack`'s way is going to change soon, and that all the optimized machinery to build the query parameters is readily available in `Queryable`, I would stick with the current implementation.

What do you think?